src/intl/context.js

```diff
@@ -1,16 +1,22 @@
-import FTLRuntimeParser from '../ftl/entries/parser';
 import { format } from './resolver';
 
+const { classes: Cc, interfaces: Ci, utils: Cu } = Components;
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "L20nParser",
+                                  "resource://gre/modules/L20nParser.jsm");
+
 export class MessageContext {
-  constructor(lang, { functions = {} } = {}) {
-    this.lang = lang;
+  constructor(locales, { functions = {}, args = {} } = {}) {
+    this.locale = Array.isArray(locales) ? locales[0] : locales;;
     this.functions = functions;
+    this.args = args;
```

Comment:
I'd like to set the right expectations.  If a dev set ctx.args = { name: 'Mary' } and than the name changes, is it understood that they also need to manually reformat all messages?

If we don't need this immediately, I'd leave it out.  It's easy to add in this form in the future, but we might end up doing something more for it, too.